packages/ibm-products/src/components/Datagrid/types/index.ts

```diff
@@ -188,7 +189,17 @@ interface DataGridTableState
 }
 
 export interface DataGridTableInstance<T extends object = any>
-  extends TableInstance<T> {}
+  extends Omit<TableInstance<T>, 'state'>,
+    Partial<UsePaginationInstanceProps<any>> {
+  shouldDisableSelectRow?: (...args) => void;
```

Comment:
Looks like the expected value of `shouldDisableSelectRow` is of type boolean. Should we add that as well here?